pkg/operator/encryption/key_controller.go

```diff
@@ -0,0 +1,350 @@
+package encryption
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const encWorkKey = "key"
+
+// encryptionKeyController creates new keys if necessary. It
+// * watches
+//   - secrets in openshift-config-managed
+//   - pods in target namespace
+//   - secrets in target namespace
+// * computes a new, desired encryption config from encryption-config-<revision>
+//   and the existing keys in openshift-config-managed.
+// * derives from the desired encryption config whether a new key is needed due to
+//   - encryption is being enabled via the API or
+//   - a new to-be-encrypted resource shows up or
+//   - the EncryptionType in the API does not match with the newest existing key or
+//   - based on time (once a week is the proposed rotation interval).
```

Comment:
and due to a new external reason - https://github.com/openshift/cluster-kube-apiserver-operator/pull/584/commits/0a5781a810336d29783edb1580e4783e61bf1cbb#diff-b44a5c1483fa3b6714560797ab0f1b4bR288

Reply:
fixed